module/VuFindSearch/src/VuFindSearch/Backend/WorldCat2/Connector.php

```diff
@@ -0,0 +1,211 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  WorldCat
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFindSearch\Backend\WorldCat2;
+
+use Exception;
+use Laminas\Http\Client\Exception\RuntimeException as ExceptionRuntimeException;
+use Laminas\Http\Exception\InvalidArgumentException;
+use Laminas\Http\Exception\RuntimeException;
+use Laminas\Http\Response;
+use Laminas\Log\LoggerAwareInterface;
+use Laminas\Session\Container;
+use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+use League\OAuth2\Client\Provider\GenericProvider;
+use VuFind\Log\LoggerAwareTrait;
+use VuFindSearch\ParamBag;
+
+/**
+ * Class for accessing OCLC WorldCat search API v2.
+ *
+ * @category VuFind
+ * @package  WorldCat
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Connector implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    /**
+     * API base URL
+     *
+     * @var string
+     */
+    protected $baseUrl;
+
+    /**
+     * Constructor
+     *
+     * @param \Laminas\Http\Client $client       An HTTP client object
+     * @param GenericProvider      $authProvider OAuth2 provider
+     * @param Container            $session      Session container for persisting data
+     * @param array                $options      Additional config settings
+     */
+    public function __construct(
+        protected \Laminas\Http\Client $client,
+        protected GenericProvider $authProvider,
+        protected Container $session,
+        protected array $options = []
+    ) {
+        if (empty($options['base_url'])) {
+            throw new \Exception('base_url setting is required');
+        }
+        $this->baseUrl = $options['base_url'];
+    }
+
+    /**
+     * Get an OAuth2 token.
+     *
+     * @return string
+     * @throws IdentityProviderException
+     */
+    protected function getToken(): string
+    {
+        return $this->authProvider->getAccessToken('client_credentials', ['scope' => 'wcapi'])->getToken();
```

Comment:
Sounds like a question for @librarywebchic. I'm not sure where this code originally came from, but I'm sure she can clarify whether or not it will remain valid going forward.

Reply:
wcapi is sort of an overarching scope that gets you all the others. if you want to be more precise you'd only ask for the ones you are using which I would guess would be wcapi:view_bib and wcapi:view_institution_holdings

Reply:
Thanks, @librarywebchic! Sounds like it's probably okay to leave this as-is, but we could certainly make this configurable in future if that would be valuable.